src/core/server/config/schema/index.ts

```diff
@@ -146,7 +147,7 @@ function siblingRef<T>(key: string): SiblingReference<T> {
   return new SiblingReference(key);
 }
 
-function conditional<A, B, C>(
+function conditional<A extends SchemaLike, B, C>(
```

Comment:
__issue:__ hmm, I don't really like the signature of `SchemaLike`: `string | number | boolean | object | null | Schema | SchemaMap` since it exposes internal `Schema` and `SchemaMap` to the outside. 

At this stage I'd rather prefer if we have something like `export type ConditionalTypeValue = string | number | boolean | object | null;` inside of `conditional_type.ts` that is re-exported in `src/core/server/config/schema/types/index.ts` and used here.

What do you think @weltenwort ?

Reply:
Good catch, fixed in the latest commit - thanks!